core/extract/extractor.py

```diff
@@ -207,8 +208,11 @@ def save_logs(
             connections_file.close()
 
         # Save the replacements
+        if self.config['log_location']:
+            copy_replacements = replacements
+        else:
+            copy_replacements = self.get_copy_replacements()
```

Comment:
How would get_copy_replacements work if no endpoint and username is provided? See get_copy_replacement definition:
        cluster_object = util.cluster_dict(self.config["source_cluster_endpoint"])
        end_time = self.config["end_time"]
        start_time = self.config["start_time"]
        redshift_user = self.config["master_username"]

Reply:
It will attempt to retrieve self.config[source_cluster_endpoint], but if its not given, it will fail right?